dagfactory/_yaml.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+import os
+
+import yaml
+
+from .utils import cast_with_type
+
+
+def load_yaml_file(file_path: str) -> dict[str, any]:
+    """
+    Load a YAML file into a dictionary.
+    """
+
+    def __join(loader: yaml.FullLoader, node: yaml.Node) -> str:
+        seq = loader.construct_sequence(node)
+        return "".join([str(i) for i in seq])
+
+    def __or(loader: yaml.FullLoader, node: yaml.Node) -> str:
+        seq = loader.construct_sequence(node)
+        return " | ".join([f"({str(i)})" for i in seq])
+
+    def __and(loader: yaml.FullLoader, node: yaml.Node) -> str:
+        seq = loader.construct_sequence(node)
+        return " & ".join([f"({str(i)})" for i in seq])
+
+    yaml.add_constructor("!join", __join, yaml.FullLoader)
+    yaml.add_constructor("!or", __or, yaml.FullLoader)
+    yaml.add_constructor("!and", __and, yaml.FullLoader)
```

Comment:
We may need to reconsider this because this custom tag may not work with some linters, as we discussed in https://github.com/astronomer/dag-factory/pull/433

Reply:
Now is the time for us to review this strategy!

Reply:
I logged this ticket so we can further discuss:
https://github.com/astronomer/dag-factory/issues/517